inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -1932,36 +1893,115 @@ void MKLDNNInterpolateNode::initSupportedPrimitiveDescriptors() {
         supportedPrimitiveDescriptors.push_back({config, implDetail});
     };
 
-    auto channels = getInputShapeAtPort(DATA_ID).getRank() > 1 ? getInputShapeAtPort(DATA_ID).getStaticDims()[1] : 1;
+    const auto &dataMinDims = getInputShapeAtPort(DATA_ID).getMinDims();
+    bool isBlkApplied = getInputShapeAtPort(DATA_ID).getRank() > 1 && dataMinDims[1] != Shape::UNDEFINED_DIM && dataMinDims[1] > 1;
 
-    if (!mayiuse(cpu::x64::sse41) || mode == InterpolateMode::linear) {
+    if (!mayiuse(cpu::x64::sse41) || interpAttrs.mode == InterpolateMode::linear) {
         pushDesc(LayoutType::ncsp, ref);
     } else {
         // blk and by_channel JIT kernel on sse41 or above machine
-        if (getInputShapeAtPort(DATA_ID).getRank() == 4 || (getInputShapeAtPort(DATA_ID).getRank() == 5 && mode != InterpolateMode::cubic)) {
+        if (getInputShapeAtPort(DATA_ID).getRank() == 4 || (getInputShapeAtPort(DATA_ID).getRank() == 5 && interpAttrs.mode != InterpolateMode::cubic)) {
             if (mayiuse(cpu::x64::avx512_common)) {
                 pushDesc(LayoutType::nspc, jit_avx512);
-                if (channels != 1)
+                if (isBlkApplied)
                     pushDesc(LayoutType::nCsp16c, jit_avx512);
             } else if (mayiuse(cpu::x64::avx2)) {
                 pushDesc(LayoutType::nspc, jit_avx2);
-                if (channels != 1)
+                if (isBlkApplied)
                     pushDesc(LayoutType::nCsp8c, jit_avx2);
             } else {
                 pushDesc(LayoutType::nspc, jit_sse42);
-                if (channels != 1)
+                if (isBlkApplied)
                     pushDesc(LayoutType::nCsp8c, jit_sse42);
             }
         }
 
         // planar for 1.ref on machine without sse41(if no sse41, canFuse() is false). 2.JIT kernel for f32 && avx2(gather).(with fuse)
-        if (mayiuse(cpu::x64::avx2) && inputPrec == Precision::FP32) {
+        if (mayiuse(cpu::x64::avx2) && inputPrecision == Precision::FP32) {
             pushDesc(LayoutType::ncsp, jit_avx2);
         }
     }
 }
 
-void MKLDNNInterpolateNode::createPrimitive() {
+bool MKLDNNInterpolateNode::needShapeInfer() const {
+    if (shapeCalcMode == InterpolateShapeCalcMode::scales) {
+        if (scalesShapeInfer.empty()) {
+            return true;
+        }
+        static constexpr float epsilon = 1.0e-6f;
+        const float *scales = reinterpret_cast<const float *>(getParentEdgesAtPort(SCALES_ID)[0]->getMemory().GetPtr());
+        for (size_t i = 0; i < scalesShapeInfer.size(); i++) {
+            if (std::abs(scalesShapeInfer[i] - scales[i]) > epsilon) {
```

Comment:
Why we are using epsilon here? Does it guarantee that we will always have the same output shape in case diff between new and old scales is within the bounds of 1.0e-6f?

Reply:
replaced with `!=`